packages/client/hmi-client/src/components/responsive-matrix/matrix.vue

```diff
@@ -706,21 +697,91 @@ export default {
 			return true;
 		},
 
+		/**
+		 * Given a 1D cell array with cells pushed in from left-to-right and top-to-bottom, iterate over
+		 * the indexes of all the cells in the selection. A negative value in the selection instructs the
+		 * function to use the topmost/leftmost row/col in the case of a start value or the bottommost/rightmost
+		 * row/col in the case of end value. A callback is executed with the index as a parameter for each
+		 * index. If the callback returns true then stop iterating (as a mechanism to support early
+		 * termination). The function returns a boolean indicating whether early termination was triggered.
+		 * @param {SelectedCell} selectedCell
+		 * @param {(idx: number) => boolean | void} cb
+		 */
+		iterateCellIndexes(selectedCell: SelectedCell, cb: (idx: number) => boolean | void) {
+			const { START_ROW, END_ROW, START_COL, END_COL } = SelectedCellValue;
+
+			const startCol = selectedCell[START_COL] > -1 ? selectedCell[START_COL] : 0;
+			const endCol = selectedCell[END_COL] > -1 ? selectedCell[END_COL] : this.numRows - 1;
+			const startRow = selectedCell[START_ROW] > -1 ? selectedCell[START_ROW] : 0;
+			const endRow = selectedCell[END_ROW] > -1 ? selectedCell[END_ROW] : this.numRows - 1;
+
+			const numSelectedRows = endRow - startRow + 1;
+			const numSelectedCols = endCol - startCol + 1;
+			const numSelectedElements = numSelectedRows * numSelectedCols;
+
+			for (let i = 0; i < numSelectedElements; i++) {
+				const col = startCol + (i % numSelectedCols);
+				const row = startRow + Math.floor(i / numSelectedCols);
+				const idx = row * this.numRows + col;
+
+				// if cb returns true then exit early
+				if (cb(idx)) return true;
+			}
+
+			return false;
+		},
```

Comment:
This is difficult to understand...

Reply:
ill try to add some more comments, see if that improves things